### DISCUSSION TITLE
Is there other way to modify --bs-nav-pills-link-active-bg: var(--bs-white); which show in official site demo?

### DISCUSSION THREAD

**Discussion by chiahao** [twbs/bootstrap#39391] [Q&A] [answered]
Where to put sass to override the the nav-pills-link-active?
I've tried put

```sass
// 2. Include any default variable overrides here
--bs-nav-pills-link-active-bg: var(--bs-white);

// 3. Include remainder of required Bootstrap stylesheets (including any separate color mode stylesheets)
```

or 

```sass
// 2. Include any default variable overrides here
$nav-pills-link-active-bg: var(--bs-white);

// 3. Include remainder of required Bootstrap stylesheets (including any separate color mode stylesheets)
```

neither works.

Do I misunderstand the 'use sass to customize'?
Should I just use css to override?
Thanks for any help?

**Comment by julien-deramond** [comment, marked as answer]
Hey @chiahao 
For the Sass override, you got it well. Maybe you have some missing imports. Here's a working example:

```scss
@import "../node_modules/bootstrap/scss/functions";

$nav-pills-link-active-bg: var(--bs-white);

// Include remainder of required Bootstrap stylesheets (including any separate color mode stylesheets)
@import "../node_modules/bootstrap/scss/variables";
@import "../node_modules/bootstrap/scss/variables-dark";

// Some regular imports, but pick whatever you want
@import "../node_modules/bootstrap/scss/maps";
@import "../node_modules/bootstrap/scss/mixins";
@import "../node_modules/bootstrap/scss/root";
@import "../node_modules/bootstrap/scss/utilities";
@import "../node_modules/bootstrap/scss/reboot";
@import "../node_modules/bootstrap/scss/images";
@import "../node_modules/bootstrap/scss/containers";
@import "../node_modules/bootstrap/scss/grid";
@import "../node_modules/bootstrap/scss/helpers";*/
@import "../node_modules/bootstrap/scss/utilities/api";
@import "../node_modules/bootstrap/scss/nav";
```

For the custom property one, you have to define it at the end to override the existing rule by doing:

```scss
.nav-link {
  --bs-nav-pills-link-active-bg: var(--bs-white);
}
```

**Reply by chiahao** [reply]
Thank you for reply.
By the way, which method is more elegant or standard or efficient?  
I have some kind of myth that overriding in sass is better?  
Is that right?

**Reply by julien-deramond** [reply]
Based on our current architecture in Bootstrap, I'll go with Sass since you seem to be able to have a custom build.
Our custom properties are built from Sass values. So for now, I'd say it's safer because maybe the `.nav-link` implementation could change in the future (not planned but could happen). If it changes, you won't know. With the Sass approach, you're pretty much sure that we will reuse this variable whatever the implementation is, and if this variable becomes deprecated, it's going to be mentioned in the migration guide or directly when you build your Sass files with a warning message.
